ci/docker/conda-python-substrait.dockerfile

```diff
@@ -0,0 +1,35 @@
+ARG repo
+ARG arch
+ARG python=3.9
+
+FROM ${repo}:${arch}-conda-python-${python}
+
+COPY ci/conda_env_python.txt \
+     ci/conda_env_sphinx.txt \
+     /arrow/ci/
+RUN mamba install -q -y \
+        --file arrow/ci/conda_env_python.txt \
+        --file arrow/ci/conda_env_sphinx.txt \
+        $([ "$python" == "3.7" ] && echo "pickle5") \
+        python=${python} \
+        nomkl && \
+    mamba clean --all
+
+# XXX The GCS testbench was already installed in conda-cpp.dockerfile,
+# but we changed the installed Python version above, so we need to reinstall it.
+COPY ci/scripts/install_gcs_testbench.sh /arrow/ci/scripts
+RUN /arrow/ci/scripts/install_gcs_testbench.sh default
```

Comment:
I think this should not be necessary as we have a `conda-python` image for each python version, so GCS would have been installed with the same version as `ARG python`.
```suggestion
```